packages/core/src/test/awsService/cloudWatchLogs/commands/tailLogGroup.test.ts

```diff
@@ -97,10 +97,15 @@ describe('TailLogGroup', function () {
         cloudwatchSettingsSpy = sandbox.stub(CloudWatchLogsSettings.prototype, 'get').callsFake(() => {
             return 1
         })
-        await tailLogGroup(registry, testSource, codeLensProvider, {
+
+        //The mock stream doesn't 'close', causing tailLogGroup to not return. If we `await`, it will never resolve.
+        //Run it in the background and sleep, giving the command time to process the session update frames.
+        void tailLogGroup(registry, testSource, codeLensProvider, {
             groupName: testLogGroup,
             regionName: testRegion,
         })
+        await new Promise((r) => setTimeout(r, 500))
```

Comment:
instead of doing a sleep can we do a waitUntil something happens?

Reply:
Currently, TailLogGroup doesn't expose anything that we can wait on that tells us that we have done the work required to do our test assertions. 

We could possibly emit an event each time the command processes a new Stream frame. Then in the test we would waitUntil the event is emitted `X` times. 

So far the actual feature hasn't required that, but I can look into adding that for test purposes.

Reply:
Was able to use waitUntil by waiting on the LiveTailRegistry to see when there's a session in there (signals session started).

Then we can wait until the content in the document matches the expected. This is does both: validating the content of the doc, and waiting until the stream is fully processed.